packages/libs/error-stack/src/compat/anyhow.rs

```diff
@@ -43,30 +44,43 @@ impl<T> IntoReportCompat for core::result::Result<T, AnyhowError> {
         match self {
             Ok(t) => Ok(t),
             Err(anyhow) => {
+                #[allow(unused_mut)]
+                let mut source: Box<[Frame]> = Box::new([]);
+
+                #[cfg(all(nightly, feature = "std"))]
+                if let Some(backtrace) = anyhow
+                    .chain()
+                    .all(|error| error.backtrace().is_none())
+                    .then(Backtrace::capture)
+                {
+                    let frame = Frame::from_attachment(backtrace, Location::caller(), source);
+                    source = Box::new([frame]);
+                }
+
+                #[cfg(feature = "spantrace")]
+                {
+                    let frame = Frame::from_attachment(
+                        tracing_error::SpanTrace::capture(),
+                        Location::caller(),
+                        source,
+                    );
+                    source = Box::new([frame]);
+                }
+
```

Comment:
after some more thinking, we could also collect backtraces and spantraces and "reorder" their output (I have a rough idea as to how you could do that), making it possible to use attach() and making Context always the root.

Tho this will make the test checking for messages a bit more complicated.